### PR TITLE
Gogs with OAuth sidecar

### PR DESCRIPTION
**Pull request by InfoSec812**
Reconfigure Gogs to have a oauth-proxy sidecar to integrate authentication with OpenShift.

Replaces PR #32

### CODE REVIEW

**Comment by makentenza** [comment]
@InfoSec812 I can see again changes around SonarQube.... Could you review your commit history and remove that part?

**Comment by InfoSec812** [comment]
WAY too much effort to back OUT, not you... Sorry for the typo...

**Comment by InfoSec812** [comment]
@makentenza The reason was that it was conflicting with commits from other PRs which got merged. That has now been resolved and I have added it back.

File: templates/gogs/template.yaml [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+    loadBalancer: {}
+- apiVersion: v1
+  kind: Route
+  metadata:
+    annotations:
+      description: Route for OAuth authenticated HTTPS service
+    creationTimestamp: null
+    labels:
+      app: ${APPLICATION_NAME}
+    name: ${APPLICATION_NAME}
```
Comment (makentenza) [inline templates/gogs/template.yaml]: Duplicated

File: templates/gogs/template.yaml [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+    wildcardPolicy: None
+- apiVersion: v1
+  kind: Route
+  metadata:
+    annotations:
+      description: Route for application's https service.
+    creationTimestamp: null
+    labels:
+      app: ${APPLICATION_NAME}
+    name: ${APPLICATION_NAME}
```
Comment (makentenza) [inline templates/gogs/template.yaml]: Duplicated

**Comment by makentenza** [comment]
Great, thanks @InfoSec812. Plase solve the duplicate name in the Routes and I will merge it.

File: templates/gogs/template.yaml (hunk 2 above)
Comment (InfoSec812) [reply templates/gogs/template.yaml]: Resolved

File: templates/gogs/template.yaml [hunk 3]
```diff
@@ -0,0 +1,548 @@
...
+  spec:
+    port:
+      targetPort: 3000-tcp
+    tls:
+      termination: edge
+    to:
+      kind: Service
+      name: ${APPLICATION_NAME}
```
Comment (makentenza) [inline templates/gogs/template.yaml]: Service ${APPLICATION_NAME} does not exist. It must point to ${APPLICATION_NAME}-server

Comment (makentenza) [inline templates/gogs/template.yaml]: Service ${APPLICATION_NAME} does not exist. It must point to ${APPLICATION_NAME}-server

**Comment by makentenza** [comment]
@InfoSec812 I'm also getting the following error while accesing through the proxy with OAuth support: 

`{"error":"invalid_request","error_description":"The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed.","state":"f3998ba97eeffe5c11dad6f5c81e9046:/"}`

**Comment by sherl0cks** [comment]
This seems to be a dead PR. Anyone opposed to closing it?

**Comment by sherl0cks** [comment]
Hearing nothing, I am closing. We can always re open

**Pull request closed without merging**